app/models/course/assessment/question/text_response.rb

```diff
@@ -51,12 +68,30 @@ def initialize_duplicate(duplicator, other)
     copy_attributes(other)
     associate_duplicated_skills(duplicator, other)
 
-    self.solutions = duplicator.duplicate(other.solutions)
+    if comprehension_question?
+      self.groups = duplicator.duplicate(other.groups)
+    else
+      self.solutions = duplicator.duplicate(other.solutions)
+    end
   end
 
   private
 
   def validate_grade
-    errors.add(:maximum_grade, :invalid_grade) if solutions.any? { |s| s.grade > maximum_grade }
+    if comprehension_question?
+      validate_grade_comprehension
+    else
+      errors.add(:maximum_grade, :invalid_grade) if solutions.any? { |s| s.grade > maximum_grade }
+    end
+  end
+
+  def validate_grade_comprehension
+    groups.each do |group|
+      group.points.each do |point|
+        point.solutions.each do |s|
+          return errors.add(:maximum_grade, :invalid_grade) if s.grade > maximum_grade
```

Comment:
Since you validate that:
 - In Groups - `maximum_group_grade <= question.maximum_grade`
 - In Points - `maximum_point_grade <= group.maximum_group_grade`
 - In solution - `solution.grade <= point.maximum_point_grade`

This has been checked for already right?

Reply:
Yup, this has already been implicitly checked in Groups, Points and Solutions.
I added this to make comprehension Questions (`validate_grade_comprehension`) consistent with non-comprehension Questions (`validate_grade`).
Nonetheless, `validate_grade_comprehension` can always be removed.

Reply:
Can suggest to remove, because validations run on every save, and we don't want to be double checking this.

Reply:
Will do.